docs/docs/noir/concepts/data_types/arrays.md

```diff
@@ -251,3 +251,23 @@ fn main() {
 }
 
 ```
+
+### as_str
+
+Converts a byte array of type `[u8; N]` to a string. Note that this performs no UTF-8 validation -
+the given array is interpreted as-is as a string.
```

Comment:
Given this description, would it be more accurate to use [`from_utf8_unchecked`](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_unchecked) and/or rename `as_str_lossy` to `array_to_str_unchecked`?

Reply:
I ended up renaming `as_str` to `as_str_unchecked`.

The comptime implementation still uses to_string_lossy since we can't use `from_utf8_unchecked` since unsafe is banned in the frontend. To fix this we'd need to change the interpreter to use byte arrays for strings instead of `String` values but we can fix this in a later PR I think.

Reply:
Ok, can you make an issue for this? I'd like to keep track of cases where the interpreter's behavior diverges from runtime.